src/main/java/org/opensearch/performanceanalyzer/rca/store/rca/OldGenRca.java

```diff
@@ -250,4 +250,61 @@ public double readMin() {
             return Double.NaN;
         }
     }
+
+    /**
+     * Sliding window to check the max/min olg gen usage within a given time frame
+     *
+     * @param isMinSlidingWindow true if the sliding window is for min usage, false for max usage
+     *     Provides a more general framework than MinOldGenSlidingWindow as this sliding window can
+     *     be implemented as minSlidingWindow or maxSlidingWindow depending on the need.
+     */
+    public static class MinMaxSlidingWindow extends SlidingWindow<SlidingWindowData> {
+        boolean isMinSlidingWindow;
+
+        public MinMaxSlidingWindow(
+                int SLIDING_WINDOW_SIZE_IN_TIMESTAMP,
+                TimeUnit timeUnit,
+                boolean isMinSlidingWindow) {
+            super(SLIDING_WINDOW_SIZE_IN_TIMESTAMP, timeUnit);
+            this.isMinSlidingWindow = isMinSlidingWindow;
+        }
+
+        @Override
+        public void next(SlidingWindowData e) {
+            boolean pollFirstCondition;
```

Comment:
This variable `pollFirstCondition` is not used. We can remove it

Reply:
removed